capplets/appearance/appearance-main.c

```diff
@@ -56,6 +56,8 @@ init_appearance_data (int *argc, char ***argv, GOptionContext *context)
 
   data->filechooser_settings = g_settings_new (FILECHOOSER_SCHEMA);
   data->interface_settings = g_settings_new (INTERFACE_SCHEMA);
+  /*Loading this unconditionally may be cheaper than another system() call*/
+  data->interface_gnome_settings = g_settings_new (INTERFACE_GNOME_SCHEMA);
```

Comment:
you could remove the comment or guard with a `! GDK_IS_X11_DISDPLAY()` call

Reply:
BTW if you wanted this schema to be optional (in case it makes any sense to have it optional), you could do something like this:
```c
  GSettingsSchemaSource *source = g_settings_schema_source_get_default ();

  if (source)
  {
    GSettingsSchema *schema = g_settings_schema_source_lookup (source, INTERFACE_GNOME_SCHEMA, TRUE);

    if (schema)
      data->interface_gnome_settings = g_settings_new_full (schema, NULL, NULL);
  }
```

Again, not saying you should do that, as there's no point if we have to have this schema for things to work.

Reply:
Since we no longer use system() to tell if we are under wayland or x11, we no longer have to deal unconditionally with that schema. Wayfire btw does indeed depend on it as wayfire (and probably a lot of other non-GNOME/non-KDE compositors) use the GNOME gsettings values.

Reply:
OK so no point in making things more complicated than needed

Reply:
Also just confirmed that at least Debian's GTK3 packages directly depend on gsettings-desktop-schemas, which is the package containing the GNOME interface schema

Reply:
Hum, [I don't see that](https://packages.debian.org/bookworm/libgtk-3-0), where do you see it?  Not saying it can be installed without, but it would be an indirect dependency)

Reply:
https://packages.debian.org/bookworm/libgtk-3-0

Reply:
Anyway, latest commit ensures the absence of the GNOME schemas should not crash the control center even when running under wayland, we just don't get theming of apps under wayland that follow the GNOME gsettings values. We could pop up a dialog to warn the user in such cases though

Reply:
> https://packages.debian.org/bookworm/libgtk-3-0

I don't see the dep here?  Anyway, as you now only use them if they are present the question is moot.